src/safe_op.hpp

```diff
@@ -29,9 +29,16 @@
 
 #include <limits>
 #include <stdexcept>
+#include <type_traits>
 
 #ifdef _MSC_VER
 #include <Intsafe.h>
+
+// MSVC is stupid and pollutes the global namespace with max() and min() macros
+// that break std::numeric_limits<T>::max() and min()
+#undef max
+#undef min
+
```

Comment:
We should define `NOMINMAX`, see https://stackoverflow.com/questions/4913922/possible-problems-with-nominmax-on-visual-c.

Reply:
It is actually already defined in `cmake/compilerFlags.cmake`. Although it seems it is only defined for MSVC. @D4N maybe you had issues with MinGW ?